### ISSUE TITLE
Interop with C on Windows breaks across Erlang upgrades

### ISSUE THREAD

**Issue by eteeselink** [elixir-lang/elixir#5271]
### Environment

* Elixir version (elixir -v): 

        Erlang/OTP 19 [erts-8.1] [64-bit] [smp:4:4] [async-threads:10]
        Elixir 1.3.3
* Operating system: Windows 10 Pro

### Current behavior

I tried to compile a project that depends on [https://github.com/riverrun/comeonin](Comeonin), which has a native code component. It compiled fine before with the Visual Studio Build Tools loaded, no problem.

However, in an attempt to fix an unrelated problem, I had uninstalled and installed Elixir, which also upgraded my Erlang from 8.0 to 8.1. This changed my erlang path from `c:\bin\erl8.0` to `c:\bin\erl8.1`.

So far so good, but after that, Comeonin wouldn't build anymore. I forgot to copy the exact error message, but the gist was that it passed  `C:\bin\erl8.0\erts-8.0\include` as an include path to the compiler, and not `C:\bin\erl8.1\erts-8.1\include`

It took me a while to figure out what was going on, but the final hint was on <https://github.com/elixir-lang/elixir/wiki/Interoperability-with-C#makefilewin>: It recommends a way to work around an `nmake` limitation which autogenerates a file called `Makefile.auto.win` that contains the ERTS include path.

While I had successfully uninstalled the old Erlang version, Comeonin's `Makefile.auto.win` from before the upgrade was still happily there, telling nmake to look in a nonexistent folder for include files. 

I _suspect_ that this problem is fixed simply by deleting the `Makefile.auto.win` file after the nmake invocation (both when fails and when it succeeds). I tried to reproduce it and propose a documentation change (and then file a PR to Comeonin, who followed that doc quite precisely), but I'm too much of a noob with both mix and makefiles and I got stuck trying to get Comeonin to recompile at all.

### Expected behavior

I expected Comeonin to compile just fine :-)

**Comment by josevalim** [comment]
Hi @eteeselink, that's an excellent bug report but I believe this should be filed in the Comeonin project? https://github.com/riverrun/comeonin It doesn't look like Elixir per-se could do anything to solve the issue above.

**Comment by eteeselink** [comment]
@josevalim True! However Comeonin just copied&pasted the recommended makefile from the Elixir wiki, which this is the issue tracker of. I figured starting at the source of the bug (even if it's a wiki and not a piece of code) makes most sense. However I sortof forgot that the wiki is a *wiki(, so maybe indeed it's better to involve the Comeonin guys right away.

So I did: https://github.com/riverrun/comeonin/issues/92
